context/context.go

```diff
@@ -51,6 +54,18 @@ func newPooledContext(pool contextPool) Context {
 	return &ctx{pool: pool}
 }
 
+func (c *ctx) GoContext() (xnetcontext.Context, bool) {
+	if c.goCtx == nil {
+		return nil, false
+	}
+
+	return c.goCtx, true
```

Comment:
`return c.goCtx, c.goCtx != nil`? Maybe that's getting too cute though

Reply:
too cute